### PR TITLE
Make formParameters optional for Xrm.Navigation.openForm in xrm

### PR DESCRIPTION
**Pull request by derekfinlinson**
On the [Xrm.Navigation.openForm](https://docs.microsoft.com/en-us/dynamics365/customer-engagement/developer/clientapi/reference/xrm-navigation/openform) method, formParameters are optional but were set as required.

### CODE REVIEW

File: types/xrm/index.d.ts [hunk 1]
```diff
@@ -4198,6 +4198,13 @@ declare namespace Xrm {
             confirmed: boolean;
         }
 
+        interface OpenFormResult {
+            /**
+             * Identifies the record displayed or created
+             */
+            savedEntityReference: LookupValue[]
```
Comment (daryllabar) [inline types/xrm/index.d.ts]: According to the documentation, This returns a single value, not an array.

**Comment by derekfinlinson** [comment]
That's how I originally had it but thought I'd test it first and it returned an array.

**Comment by daryllabar** [comment]
That's lovely...

**Comment by daryllabar** [comment]
@derekfinlinson I've reached out to MS.  Do you have some sample code and how it is being called to be able to test out why you're seeing an array rather than an object?

**Review by daryllabar — APPROVED** [review]
The one question I had on the [] vs the single value.  Double checked, and this is how it is functioning currently.

**Pull request merged**